### PR TITLE
Support php 8.0's mixed type

### LINKED ISSUE #154: Supporting TYPE_MIXED from php 8.0 mixed type
My thoughts:

It seems like `mixed` isn't a reserved keyword, so the approach used for `iterable`(7.1) and `object`(7.2) would be consistent with prior releases, and existing tooling would point out that there was no class with name `mixed` (if there wasn't).

In AST version 70 and below, continue returning a `ast\AST_NAME`(NAME_NOT_FQ) with the name "mixed" (preserve original case if available)

1. Make PHP 8.0 emulate php 7.x
2. Create a brand new AST version 75 where php 7 and 8 will return AST_TYPE with TYPE_MIXED

https://github.com/TysonAndre/php-ast/compare/is_mixed?targetBranch=master is a WIP branch for this that doesn't yet introduce a new version

### PR DESCRIPTION
**Pull request by TysonAndre**
Fixes #154

- We've preferred AST versions that are multiples of 10 for previous releases.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -270,6 +270,7 @@ ast\flags\TYPE_OBJECT
 ast\flags\TYPE_NULL    // php 8.0 union types
 ast\flags\TYPE_FALSE   // php 8.0 union types
 ast\flags\TYPE_STATIC  // php 8.0 static return type
+ast\flags\TYPE_MIXED  // php 8.0 mixed type
```
Comment (nikic) [inline README.md:273]: Alignment

File: ast.c [hunk 2]
```diff
@@ -580,6 +584,7 @@ static inline void ast_name_to_zval(zend_ast *child, zend_ast *ast, zval *child_
 	if (child->attr == ZEND_NAME_NOT_FQ
 			&& ast_is_type(child, ast, i)
 			&& (type = lookup_builtin_type(zend_ast_get_str(child)))
+			&& type != IS_MIXED || state->version >= 80
```
Comment (nikic) [inline ast.c:587]:
```suggestion
			&& (type != IS_MIXED || state->version >= 80)
```

File: README.md (hunk 1 above)
Comment (TysonAndre) [reply README.md]: Done

File: ast.c (hunk 2 above)
Comment (TysonAndre) [reply ast.c]: Done

**Pull request merged**